src/main/scala/dpla/ingestion3/preMappingReports/PreMappingReporter.scala

```diff
@@ -0,0 +1,100 @@
+package dpla.ingestion3.premappingreports
+
+import java.io.File
+
+import dpla.ingestion3.utils.Utils
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import com.databricks.spark.avro._
+
+import scala.util.Try
+
+/**
+  * This class reads in harvested data.
+  * It calls PreMappingReport to generate reports.
+  * It writes out the reports.
+  *
+  * @param inputDir Path to harvested data.
+  * @param outputDir Path to pre-mapping reports.
+  * @param sparkMasterName
+  * @param inputDataType "json", "xml"
+  */
+class PreMappingReporter(val inputDir: String,
+                             val outputDir: String,
+                             val sparkMasterName: String,
+                             val inputDataType: String) extends Serializable {
+
+  private val spark: SparkSession = {
+    val sparkConf: SparkConf =
+      new SparkConf().setAppName("PreMappingReport").setMaster(sparkMasterName)
+    SparkSession.builder().config(sparkConf).getOrCreate()
+  }
+
+  private val report = {
+    val input: DataFrame = spark.read.avro(inputDir)
+    new PreMappingReport(input, inputDataType, spark)
+  }
+
+  private val outputDirName: String = {
+    if (outputDir.endsWith("/")) outputDir.dropRight(1)
+    else outputDir
+  }
+
+  // Main entry point.
+  def writeAllReports: Try[Unit] = Try {
+
+    // If the output directory already exists it is a local path
+    // then delete it and its contents.
+    val outputFile = new File(outputDir)
+    outputFile.getParentFile.mkdirs()
+    if (outputFile.exists & !outputDir.startsWith("s3"))
+      Utils.deleteRecursively(outputFile)
+
+    // Write processed data.
+    // TODO: Determine whether we want to write out all of reports or allow
+    // user to choose which reports they want to generate.  We may also decide
+    // to eliminate some of these reports if we don't need them.
+    writeShreddedData
+    writeElasticSearchData
+    writeCardinalityReport
+    writeOpenRefineData
+    writeRecordsMissingAttributesReport
+
+    // Stop spark context.
+    spark.sparkContext.stop()
+  }
+
+  private def writeShreddedData: Unit =
```

Comment:
IntelliJ IDEA is telling me that these methods of type `Unit` are supposed to have empty parentheses; e.g. `private def writeShreddedData(): Unit`, the convention being that there are parentheses if the method has side effects.